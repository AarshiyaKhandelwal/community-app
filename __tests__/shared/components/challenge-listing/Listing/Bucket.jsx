import React from 'react';
import _ from 'lodash';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import Bucket from 'components/challenge-listing/Listing/Bucket';

const expand = jest.fn();
const loadMore = jest.fn();
const setFilterState = jest.fn();
const setSort = jest.fn();

const mockDatas = [{
  bucket: {
    sorts: ['current-phase'],
    filter: {
      status: 'ac',
    },
  },
  expanded: true,
  expand,
  challenges: [
  { id: '1', status: 'b' },
    {
      id: '2',
      status: 'a',
      technologies: 'a, b, c',
      allPhases: [{
        phaseType: 'Registration',
        phaseStatus: 'Open',
      }],
      currentPhases: ['Registration'],
      track: 'DEVELOP',
      subTrack: 'CODE',
      events: [{ eventName: 'Submit' }],
    }],
  loading: false,
  loadMore,
  setFilterState,
  setSort,
  sort: '',
}, {
  bucket: {
    sorts: ['current-phase'],
    filter: {
      status: 'ac',
    },
  },
  expanded: false,
  expand,
  challenges: [
    { id: '1', status: 'b' },
    { id: '2', status: 'a' },
    { id: '3', status: 'a' },
    { id: '4', status: 'a' },
    { id: '5', status: 'a' },
    { id: '6', status: 'a' },
    { id: '7', status: 'a' },
    { id: '8', status: 'a' },
    { id: '9', status: 'a' },
    { id: '10', status: 'a' },
    { id: '11', status: 'a' },
  ],
  loading: true,
  loadMore,
  setFilterState,
  setSort,
  sort: '',
}, {
  bucket: {
    sorts: ['current-phase'],
    filter: {
      status: 'e',
    },
  },
  expanded: false,
  expand,
  challenges: [
  ],
  loading: true,
  setFilterState,
  setSort,
  sort: '',
}];

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  _.forEach(mockDatas, (data) => {
    renderer.render((
      <Bucket {...data} />
    ));
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});

class Wrapper extends React.Component {
  componentDidMount() {}

  render() {
    return <Bucket {...this.props} />;
  }
}

const instance = TU.renderIntoDocument((<Wrapper {...mockDatas[0]} />));

test('setFilterState', () => {
  const matches = TU.findAllInRenderedTree(instance, item =>
    item && item.className && item.className.match('technology'));
  expect(matches).toHaveLength(3);
  TU.Simulate.click(matches[0]);
  expect(setFilterState).toHaveBeenCalledTimes(1);
  TU.Simulate.click(matches[1]);
  expect(setFilterState).toHaveBeenCalledTimes(2);
  TU.Simulate.click(matches[2]);
  expect(setFilterState).toHaveBeenCalledTimes(3);
});

