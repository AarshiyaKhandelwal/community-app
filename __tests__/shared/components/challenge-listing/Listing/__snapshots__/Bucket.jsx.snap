// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-challenge-listing-Listing-Bucket-___style__bucket___3jwPe"
>
  <SortingSelectBar
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "Current phase",
          "value": "current-phase",
        },
      ]
    }
    title=""
    value={
      Object {
        "label": "Current phase",
        "value": "current-phase",
      }
    }
  />
  <ChallengeCard
    challenge={
      Object {
        "allPhases": Array [
          Object {
            "phaseStatus": "Open",
            "phaseType": "Registration",
          },
        ],
        "currentPhases": Array [
          "Registration",
        ],
        "events": Array [
          Object {
            "eventName": "Submit",
          },
        ],
        "id": "2",
        "isDataScience": false,
        "prize": Array [],
        "status": "a",
        "subTrack": "CODE",
        "technologies": "a, b, c",
        "track": "DEVELOP",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <Waypoint
    bottomOffset="0px"
    fireOnRapidScroll={true}
    horizontal={false}
    onEnter={[Function]}
    onLeave={[Function]}
    onPositionChange={[Function]}
    topOffset="0px"
  />
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="src-shared-components-challenge-listing-Listing-Bucket-___style__bucket___3jwPe"
>
  <SortingSelectBar
    onSelect={[Function]}
    options={
      Array [
        Object {
          "label": "Current phase",
          "value": "current-phase",
        },
      ]
    }
    title=""
    value={
      Object {
        "label": "Current phase",
        "value": "current-phase",
      }
    }
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "6",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "11",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "3",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "4",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "5",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "2",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "7",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "8",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "9",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCard
    challenge={
      Object {
        "id": "10",
        "status": "a",
      }
    }
    onTechTagClicked={[Function]}
    sampleWinnerProfile={undefined}
  />
  <ChallengeCardPlaceholder
    id={0}
  />
  <ChallengeCardPlaceholder
    id={1}
  />
  <ChallengeCardPlaceholder
    id={2}
  />
  <ChallengeCardPlaceholder
    id={3}
  />
  <ChallengeCardPlaceholder
    id={4}
  />
  <ChallengeCardPlaceholder
    id={5}
  />
  <ChallengeCardPlaceholder
    id={6}
  />
  <ChallengeCardPlaceholder
    id={7}
  />
  <button
    className="src-shared-components-challenge-listing-Listing-Bucket-___style__view-more___2UekL"
    onClick={[Function]}
  >
    View more challenges
  </button>
</div>
`;

exports[`Matches shallow shapshot 3`] = `null`;
